### ISSUE TITLE
The behavior of new Set([]) not work in same way caused some bug

### ISSUE THREAD

**Issue by bolasblack** [google/lovefield#81]
My safari version is `8.0.8 (10600.8.9)`, my chrome version is `45.0.2454.85 (64-bit)`

I found the behavior of `new Set([])` in safari is not as same as chrome:

```javascript
// in chrome
new Set([1,2,3]).size // #=> 3

// in safari
new Set([1,2,3]).size // #=> 1
```

This issue caused a bug:

The code in [`lf.cache.Prefetcher.prototype.fetchTable_`](https://github.com/google/lovefield/blob/d0293010c01640730c18c21da9ef7cd425f24953/lib/cache/prefetcher.js#L91) initialized a set by pass in an array:

```javascript
var journal =
      new lf.cache.Journal(this.global_, lf.structs.set.create([table]));
```

And [`lf.cache.Journal` try to call `getName` from all elements in the set](https://github.com/google/lovefield/blob/master/lib/cache/journal.js#L63):

```javascript
lf.cache.Journal = function(global, scope) {
  this.scope_ = lf.structs.map.create();
  scope.forEach(function(tableSchema) {
    this.scope_.set(tableSchema.getName(), tableSchema);
  }, this);
```

Then the code threw an error said `tableSchema.getName` is not a function.

**Comment by arthurhsu** [comment]
Not likely, see https://github.com/google/lovefield/blob/master/lib/structs/set.js#L131

Lovefield uses SetPolyFill_ when Safari is detected.

**Comment by bolasblack** [comment]
@arthurhsu Sorry for my indiscretion, I will waiting for the new release :P

**Comment by bolasblack** [comment]
@arthurhsu Hey, I found `goog.userAgent.product.SAFARI` in `NSWebView` is not `true`, but the issue is also exists.

**Comment by arthurhsu** [comment]
The fix is under way, stay tuned.

**Comment by arthurhsu** [comment]
Fixed by addd6bc6dca562b78cdc64e0fe1fd67be998ff99

**Comment by bolasblack** [comment]
@arthurhsu Awesome! :+1:
